src/main/java/org/jenkinsci/plugins/pubsub/Message.java

```diff
@@ -205,6 +231,44 @@ public T setEventName(Enum name) {
     }
 
     /**
+     * Get the Jenkins instance URL of the master from which the event was published.
+     *
+     * @return The Jenkins instance identity.
+     * @see #getJenkinsInstanceId()
+     */
+    public String getJenkinsInstanceUrl() {
+        return get(EventProps.Jenkins.jenkins_instance_url);
+    }
+
+    /**
+     * Get the Jenkins instance identity of the master from which the event was published.
+     * <p>
+     * <strong>Note</strong> that this is not automatically added to every event since the
+     * identity key is quite large, adding a lot of weight to each event. To add the identity
+     * to all messages, simply implement a {@link MessageEnricher} and use it to call
+     * {@link #setJenkinsInstanceId()}.
+     * <p>
+     * Maybe {@link #getJenkinsInstanceUrl()} will do the trick for you
+     * in terms of working out the origin of an event.
+     * 
+     * @return The Jenkins instance identity.
+     * @see #getJenkinsInstanceUrl()
+     */
+    public String getJenkinsInstanceId() {
+        return get(EventProps.Jenkins.jenkins_instance_id);
+    }
+
+    /**
+     * Set the Jenkins instance identity of the master from which the event was published.
+     */
+    public T setJenkinsInstanceId() {
```

Comment:
We should let enrichers compute instance id and set that instead of precomputed one. Something like `public T setJenkinsInstanceId(@Nonnull String instanceId)`. For example, blueocean computes it's instance id as sha256 hash. Doing this also gets rid of static initializer block.